composer/callbacks/memory_monitor.py

```diff
@@ -101,6 +141,13 @@ def after_train_batch(self, state: State, logger: Logger):
             return
 
         memory_report = _get_memory_report(self.memory_keys)
+        if self.dist_aggregate_batch_interval is not None and state.timestamp.batch.value % self.dist_aggregate_batch_interval == 0:
+            dist_memory_report = {}
+            for (mem_stat, val) in memory_report.items():
+                dist_memory_report[mem_stat + '_avg'] = reduce_value(val, model_device, 'avg')
+                dist_memory_report[mem_stat + '_min'] = reduce_value(val, model_device, 'min')
+                dist_memory_report[mem_stat + '_max'] = reduce_value(val, model_device, 'max')
+            memory_report.update(dist_memory_report)
```

Comment:
Can you update docstring to include a comment saying these are provided if the interval is set?

Reply:
The `if` gating at [144](https://github.com/mosaicml/composer/pull/3042/files/14227f30492e283521d0da9ad8553a8eca717d45#diff-529d790253a002590be026ce3b6b59a345a08057be280fa3651ae055a7c495daR144) gives it away \s


yeah I'll add it.

Reply:
https://github.com/mosaicml/composer/pull/3042/commits/278af0549522a4ebe99bbf4be4c1b81175638440